### PR TITLE
JS Components: Convert utility functions to TypeScript

### PR DESCRIPTION
**Pull request by manzoorwanijk**
#### Changes proposed in this Pull Request:
* This PR converts utility functions in `js-packages/components/tools` and `js-packages/components/lib` to TypeScript.
* It also converts their unit tests to TS

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `jetpack test js-packages/components js`
* Confirm that the tests pass

### CODE REVIEW

**Review by dhasilva — DISMISSED** [review]
Looks good :+1:

File: projects/js-packages/components/tools/jp-redirect/index.ts [hunk 1]
```diff
@@ -10,15 +13,11 @@
...
- * @param {string} args.site - URL of the current site. Will default to the value of jetpack_redirects.currentSiteRawUrl, if available. Used to fill in the `[site]` placeholder in the target.
- * @param {string} args.path - Additional path to be appended to the URL. Used to fill in the `[path]` placeholder in the target.
- * @param {string} args.query - Query parameters to be added to the final destination URL. should be in query string format (e.g. 'key=value&foo=bar').
- * @param {string} args.anchor - Anchor to be added to the URL. Must be a single string. Example: `section1`.
+ * @param {GetRedirectUrlArgs} args - Additional arguments to build the url.  This is not a complete list as any argument passed here will be sent to as a query parameter to the Redirect server. These parameters will not necessarily be passed over to the final destination URL. If you want to add a parameter to the final destination URL, use the `query` argument.
```
Comment (retrofox) [inline projects/js-packages/components/tools/jp-redirect/index.ts]: could you split this comment into a few lines?

Comment (manzoorwanijk) [reply projects/js-packages/components/tools/jp-redirect/index.ts]: Done, thank you.

**Review by samiff — APPROVED** [review]

**Comment by manzoorwanijk** [comment]
r250218-wpcom

**Comment by bindlegirl** [comment]
@manzoorwanijk I see you merged a diff but there is also a related diff that's waiting here: D85919-code 
Could you please check it and close it if it's not needed?

**Comment by manzoorwanijk** [comment]
> I see you merged a diff but there is also a related diff that's waiting here: D85919-code
> Could you please check it and close it if it's not needed?

That's not needed. How do I close it?

**Pull request merged**